### PR TITLE
carl: init at 0.3.1

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Things done

Slightly slower than the util-linux version of `cal` and the tests fail, but it's an interesting project to keep an eye on

```
hyperfine --warmup 500 --shell=none /nix/store/6z0c5iiibq6j53ywp0i7ipc6cpdbkxcp-carl-0.3.1/bin/carl /nix/store/4wj2h8f0npnf7slsdgy1hy985pxpj4lx-util-linux-2.39.4-bin/bin/cal 2>/dev/null
Benchmark 1: /nix/store/6z0c5iiibq6j53ywp0i7ipc6cpdbkxcp-carl-0.3.1/bin/carl
  Time (mean ± σ):       1.4 ms ±   0.2 ms    [User: 0.2 ms, System: 1.2 ms]
  Range (min … max):     1.3 ms …   3.2 ms    1664 runs
 
Benchmark 2: /nix/store/4wj2h8f0npnf7slsdgy1hy985pxpj4lx-util-linux-2.39.4-bin/bin/cal
  Time (mean ± σ):       1.4 ms ±   0.2 ms    [User: 0.9 ms, System: 0.4 ms]
  Range (min … max):     1.3 ms …   3.0 ms    2205 runs
 
Summary
  /nix/store/4wj2h8f0npnf7slsdgy1hy985pxpj4lx-util-linux-2.39.4-bin/bin/cal ran
    1.04 ± 0.18 times faster than /nix/store/6z0c5iiibq6j53ywp0i7ipc6cpdbkxcp-carl-0.3.1/bin/carl
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12495673321

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12495673321


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368115`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>carl</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368115`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>carl</li>
  </ul>
</details>

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368115`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>carl</li>
  </ul>
</details>

File: pkgs/by-name/ca/carl/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "b1rger";
+    repo = "carl";
+    rev = "v${version}";
+    hash = "sha256-+l11eP+1qKrWbZhyUJgQ8FgQ+2rncx778F5RPzCfvV4=";
+  };
+
+  doCheck = false;
```
Comment (lucasew) [inline pkgs/by-name/ca/carl/package.nix:18]: Please enable tests

Comment (MatthewCroughan) [reply pkgs/by-name/ca/carl/package.nix:18]: I've explained in the description that tests are not functional, and I reported it in an upstream issue.

Comment (lucasew) [reply pkgs/by-name/ca/carl/package.nix:18]: Seems like some kind of nix issue. I cloned and ran the test on latest master and v0.3.1 and all ran successfully.

Comment (lucasew) [reply pkgs/by-name/ca/carl/package.nix:18]: I think I found the culprit, if it works I may fix the tests for a handful of other packages too.

https://github.com/NixOS/nixpkgs/pull/368593

**Pull request merged**